File: leetcode-04/solutions/cpp/0454.cpp
class Solution {
 public:
  int fourSumCount(vector<int>& A, vector<int>& B, vector<int>& C,
                   vector<int>& D) {
    int ans = 0;
    unordered_map<int, int> map;

    for (int a : A)
      for (int b : B) map[a + b]++;

    for (int c : C)
      for (int d : D) ans += map.count(-c - d) ? map[-c - d] : 0;

    return ans;
  }
};

File: leetcode-04/solutions/python3/0454.py
class Solution:
    def fourSumCount(self, A: List[int], B: List[int], C: List[int], D: List[int]) -> int:
        dict = collections.Counter(a + b for a in A for b in B)

        return sum(dict[-c - d] for c in C for d in D)
